bin/concat_h5ad.py

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env python
+import scanpy as sc, anndata as ad, pandas as pd
+from pathlib import Path
+import argparse
+
+# empty list to hold sample datasets
+list_of_h5ad = []
+
+def read_samplesheet(samplesheet):
+    df = pd.read_csv(samplesheet)
+    return(df)
+
+# find available h5ad files and append to list
+def append_h5ad_files():
+    for path in Path(".").rglob('*.h5ad'):
+        adata = sc.read_h5ad(path.name)
+        list_of_h5ad.append(adata)
+
+# combine and write
+# combination without inner or out join, just a simple concatenation.
+def concat_h5ad(outfile):
+    combined = ad.concat(list_of_h5ad)
```

Comment:
I suggest to add `index_unique="<SAMPLE_ID>"`, as barcodes are not unique across samples, where SAMPLE_ID refers to the column in `adata.obs` that contains the sample id

Reply:
`index_unique` is how you specify the separator when joining the label with the `obs_names`, you want to associate a sample label with the obs names by passing a dict of anndata objects or with the `keys` argument.

In general I'd suggest structuring the code a bit more like I did [here](https://discourse.scverse.org/t/how-to-add-additional-metadata-on-multiple-different-single-cell-rna-files-from-different-experiments-such-as-organ-sample-number-etc/533/2?u=ivirshup)